xwiki-commons-tools/xwiki-commons-tool-xar/xwiki-commons-tool-xar-integration-tests/src/test/java/org/xwiki/tool/xar/XARMojoTest.java

```diff
@@ -211,6 +211,9 @@ void transformXML() throws Exception
 
         assertTrue(document.selectSingleNode("/xwikidoc/attachment/content") != null,
             "Insertion of attachment did not happen?");
+
+        assertTrue(document.selectSingleNode("//insertHereBlogXWikiClass") == null,
+                "Replacement of insertHereBlogXWikiClass did not happen?");
```

Comment:
Indentation has 4 chars too much.

Reply:
Is it normal to use "//" instead of the "/xwikidoc/..." used in the previous assert? Especially since it's a single node, it's probably more accurate to test for a precise location. WDYT?

Reply:
> "//" instead of the "/xwikidoc/..." used in the previous assert? Especially since it's a single node, it's probably more accurate to test for a precise location. WDYT?

Thus the test-verification is broader: it asserts that no element with that local-name are anywhere in the tree. So it makes it a more sure test!

> Indentation has 4 chars too much.

fixed.

Reply:
> Thus the test-verification is broader: it asserts that no element with that local-name are anywhere in the tree. So it makes it a more sure test!

I disagree. Adding `insertHereBlogXWikiClass` in the wrong part of the XML is not correct and the test should verify that. The test must verify it's added in the right place.

Reply:
You disagree with the fact that the verification is broader?
Or do you feel the test should verify that the replacement has happened by the existence of a particular element?

Reply:
Your transformation is defined as:

```
<transformation> 
   <file>Blog/WebHome.xml</file> 
   <action>REPLACE</action> 
   <xpath>/xwikidoc/object/insertHereBlogXWikiClass</xpath> 
   <xml>src/main/resources/Blog/BlogClass.xwikiclass</xml> 
 </transformation> 
```

So AFAIU we need to verify that `/xwikidoc/object/insertHereBlogXWikiClass` is replaced and only that part of the XML. If another part is replaced then the code is clearly wrong.

Reply:
Well, in the case of this example and example file (a test is defined by all its parts!) both are the same. Why should the test be minimal and ignore the nature of the input-data?

I guess it might be more elegant to write `<xpath>//insertHereBlogXWikiClass</xpath>` but showing by an automated test that both things are converging is actually showing that the machine works well. Another thing that this tests, is that the element has not been moved somewhere else (e.g. the insertion has taken place but the element is at some other place...).